lib/sinon/fake.js

```diff
@@ -37,70 +37,87 @@ function wrapFunc(f) {
     return proxy;
 }
 
-function fake(f) {
-    if (arguments.length > 0 && typeof f !== "function") {
-        throw new TypeError("Expected f argument to be a Function");
-    }
+function fakeClass() {
+    var promiseLib;
```

Comment:
If they always have the same interface, why bother with branching? 
```suggestion
    var promiseLib = Promise;
```